recipes_source/amx.rst

```diff
@@ -0,0 +1,108 @@
+==============================================
+Leverage Advanced Matrix Extensions
+==============================================
+
+Introduction
+============
+
+Advanced Matrix Extensions (AMX), also known as Intel® Advanced Matrix Extensions (Intel® AMX), is an extension to the x86 instruction set architecture (ISA).
```

Comment:
I realize AMX is lower level than other Intel technologies but it's still worth rationalizing to an end user in a few lines why it's interesting for them to know about AMX vs Intel compiler technologies

Reply:
Added more introduction to AMX and the benefits it can bring.